x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/WaitForIndexColorStep.java

```diff
@@ -75,17 +75,26 @@ public boolean equals(Object obj) {
 
     @Override
     public Result isConditionMet(Index index, ClusterState clusterState) {
-        String indexName = indexNamePrefix != null ? indexNamePrefix + index.getName() : index.getName();
-        IndexMetadata indexMetadata = clusterState.metadata().index(index);
-
-        if (indexMetadata == null) {
+        // check if the original (managed) index still exists
+        if (clusterState.metadata().index(index) == null) {
```

Comment:
I'm not sure why we need to ensure the original index also exists? Otherwise it seems strange to say "wait for shrink-foo to be green" that fails if "foo" doesn't exist.

I think we can just deal with the next `if` statement, since it would check either the prefixed or non-prefixed version depending on if a prefix were specified?

Reply:
Yes, that's a good point. 

My reasoning here is that we might be in an inconsistent state as the managed index was deleted but the target index isn't. The next steps would silently not execute and there will be no settings and execution state copied to the target index. 
I believe the user can end up with a GREEN target index that's not managed by ILM and doesn't have the identity of the original index (when this step is used in the `searchable_snapshot` context). 

I think we should flag this situation but I agree the messaging might be confusing